urc_platform/src/chassis_controller_wrapper.cpp

```diff
@@ -0,0 +1,47 @@
+#include "chassis_controller.hpp"
+
+namespace chassis_controller
+{
+
+ChassisControllerWrapper::ChassisControllerWrapper(const rclcpp::NodeOptions & options)
+: rclcpp::Node("chassis_controller", options)
+{
+  std::string ip_addr_server_ = declare_parameter<std::string>("ip_addr");
+  int port_ = declare_parameter<int>("port");
+  publish_encoder_ticks_frequency_ = declare_parameter<double>("frequency");
+
+  // driver.reset(new ChassisControllerDriver(ip_addr_server_, port_));
+  driver = std::make_unique<ChassisControllerDriver>(ip_addr_server_, port_);
+
+  _enc_pub = this->create_publisher<urc_msgs::msg::VelocityPair>(
+    "~/encoders",
+    1000);
+
+  timer_ = this->create_wall_timer(
+    std::chrono::duration<double>(1.0 / publish_encoder_ticks_frequency_),
+    std::bind(&ChassisControllerWrapper::publishEncoderTicks, this));
+}
+
+void ChassisControllerWrapper::publishEncoderTicks()
+{
+
+  DriveEncodersMessage ticks;
+  bool success = driver->getEncoderTicks(ticks);
+
+  if (!success) {
+    RCLCPP_ERROR(this->get_logger(), "Decoding Failed");
```

Comment:
Probably make this warning instead, doesn't warrant shutting everything down imo